components/txn_types/src/lock.rs

```diff
@@ -95,9 +98,30 @@ impl Lock {
             b.push(MIN_COMMIT_TS_PREFIX);
             b.encode_u64(self.min_commit_ts.into_inner()).unwrap();
         }
+        if self.use_async_commit {
+            b.push(ASYNC_COMMIT_PREFIX);
+            b.encode_var_u64(self.secondaries.len() as _).unwrap();
```

Comment:
If secondaries are empty, I think no need to write it.

Reply:
Sadly, we can't know whether a lock is a primary lock without comparison. If so, we can add some assertions here.

Reply:
It appends a zero that indicates the set of secondaries is empty. It's needed to parse the lock from bytes, since as you said we can't know if it's a primary here.

Reply:
Because `ASYNC_COMMIT_PREFIX` is the last byte, the zero is not necessary. However, if we add more fields in the future, it's necessary.

Reply:
Yes. We should be prepared for adding more fields in the future.